base/sharedarray.jl

```diff
@@ -142,7 +142,7 @@ file is not writable.
 `offset` allows you to skip the specified number of bytes at the
 beginning of the file.
 """
-function SharedArray{T,N}(filename::AbstractString, ::Type{T}, dims::NTuple{N,Int},
+function SharedArray{T,N}(filename::AbstractString, dims::NTuple{N,Int},
```

Comment:
This should become a new method of `SharedArray{T}` this also needs to go in Compat.jl since it's not available on old julia versions AFAICT.